src/main/java/org/junit/Assert.java

```diff
@@ -955,4 +955,96 @@ public static void assertEquals(Object[] expecteds, Object[] actuals) {
+     * @since 4.13
+     */
+    @SuppressWarnings("unchecked")
+    public static <T extends Throwable> T expectThrows(String message, ThrowingRunnable runnable) {
+        return (T) expectThrows(message, Throwable.class, runnable);
+    }
+
+    /**
+     * Asserts that {@code runnable} throws an exception of type {@code throwableClass} when
+     * executed. If it does, the exception object is returned. If it does not throw an exception, an
+     * {@link AssertionError} is thrown. If it throws the wrong type of exception, an {@code
+     * AssertionError} is thrown describing the mismatch; the exception that was actually thrown can
+     * be obtained by calling {@link AssertionError#getCause}.
+     *
+     * @param throwableClass the expected type of the exception
+     * @param runnable       A function that is expected to throw an exception when executed
+     * @return The exception thrown by {@code runnable}
+     * @since 4.13
+     */
+    public static <T extends Throwable> T expectThrows(Class<T> throwableClass, ThrowingRunnable runnable) {
+        return expectThrows(null, throwableClass, runnable);
+    }
+
+    /**
+     * Asserts that {@code runnable} throws an exception of type {@code throwableClass} when
+     * executed. If it does, the exception object is returned. If it does not throw an exception, an
+     * {@link AssertionError} is thrown with the given {@code message}. If it throws the wrong type
+     * of exception, an {@code AssertionError} is thrown describing the mismatch; the exception that
+     * was actually thrown can be obtained by calling {@link AssertionError#getCause}.
+     *
+     * @param message        the identifying message for the {@link AssertionError}
+     * @param throwableClass the expected type of the exception
+     * @param runnable       A function that is expected to throw an exception when executed
+     * @return The exception thrown by {@code runnable}
+     * @since 4.13
+     */
+    @SuppressWarnings("unchecked")
+    public static <T extends Throwable> T expectThrows(String message, Class<T> throwableClass, ThrowingRunnable runnable) {
+        try {
+            runnable.run();
+        } catch (Throwable t) {
+            try {
+                return throwableClass.cast(t);
+            } catch (ClassCastException ex) {
+                String mismatchMessage = String.format("Expected %s to be thrown, but %s was thrown",
+                        throwableClass.getSimpleName(), t.getClass().getSimpleName());
+
+                // The AssertionError(String, Throwable) ctor is only available on JDK7.
+                AssertionError assertionError = new AssertionError(mismatchMessage);
+                assertionError.initCause(t);
+                throw assertionError;
+            }
+        }
+        if (message == null) {
+            message = String.format("Expected %s to be thrown, but nothing was thrown", throwableClass.getSimpleName());
```

Comment:
I don't think we've decided formally on a line length restriction, but since we recently switched to the Google Java Style Guide, we should probably use 100

See https://google-styleguide.googlecode.com/svn/trunk/javaguide.html#s4.5-line-wrapping